### PR TITLE
Add location to dotnet proxy

### PR DESCRIPTION
**Pull request by emmawillis**

### CODE REVIEW

File: proxies/dotnet/Controllers/Location.cs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        } catch (Exception e) {
+            Response.StatusCode = 404;
+            return new { message = e.Message };
+        }
+
+        try {
+            var result = await InvokeCommand(entity, body.Command, body.IsAsync, parsedParams);
+
+            Response.StatusCode = 201;
+            if (result.CommandId != null) Response.Headers.Add("Location", $"command/{body.Command}/{result.CommandId}");
```
Comment (ajwootto) [inline proxies/dotnet/Controllers/Location.cs:78]: when would this commandID be null?

File: proxies/dotnet/Controllers/Location.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+    private string GetEntityType(object result) {
+        var type = TypeDescriptor.GetClassName(result) ?? "Object";
+        if (type.Contains("Dictionary")) {
+            type = "Object";
+        } else {
+            type = type.Split(".").Last();
+        }
+
+        return EntityTypes.Contains(type) ? type : "Void";
```
Comment (ajwootto) [inline proxies/dotnet/Controllers/Location.cs:187]: I think "Void" is supposed to represent "nothing was returned" not "we dont know what type this corresponds to". That case should be "Object"

File: proxies/dotnet/Controllers/Location.cs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+    private List<object> ParseParams(List<JToken> bodyParams) {
+        var result = new List<object>{};
+
+        foreach (var param in bodyParams) {
+            if (param["location"] != null) {
+                result.Add(GetEntity(param["location"].ToString()));
+                Console.WriteLine(Newtonsoft.Json.JsonConvert.SerializeObject(GetEntity(param["location"].ToString())));
+            } else if (param["value"] != null) {
+                if (param["value"].Type == JTokenType.Float) {
```
Comment (ajwootto) [inline proxies/dotnet/Controllers/Location.cs:111]: is this kinda like a generic JSON cast of the data to a native type?

File: proxies/dotnet/Controllers/Location.cs (hunk 2 above)
Comment (elliotCamblor) [reply proxies/dotnet/Controllers/Location.cs:187]: yeah, really not sure why this is being done. I think it shouldnt be object either tbh, object should already be covered. This method shoulld probably just return type

File: proxies/dotnet/Controllers/Location.cs (hunk 3 above)
Comment (elliotCamblor) [reply proxies/dotnet/Controllers/Location.cs:111]: yeah, this is what we spent all afternoon trying to get working yesterday

Comment (ajwootto) [reply proxies/dotnet/Controllers/Location.cs:111]: nice

File: proxies/dotnet/Controllers/Location.cs (hunk 1 above)
Comment (elliotCamblor) [reply proxies/dotnet/Controllers/Location.cs:78]: I dont think it would be

File: proxies/dotnet/Controllers/Location.cs (hunk 2 above)
Comment (elliotCamblor) [reply proxies/dotnet/Controllers/Location.cs:187]: im just gonna make it object for now to appease the enum gods

**Pull request closed without merging**